pex/common.py

```diff
@@ -213,8 +213,9 @@ def _chmod(self, info, path):
         # This magic works to extract perm bits from the 32 bit external file attributes field for
         # unix-created zip files, for the layout, see:
         #   https://www.forensicswiki.org/wiki/ZIP#External_file_attributes
-        attr = info.external_attr >> 16
-        os.chmod(path, attr)
+        if info.external_attr > 0xFFFF:
```

Comment:
This was a bug exposed by GitHub zips, they have no perm bits set (signs of MS I think). Their tarballs present no such problems.